runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatKeyboardListener.java

```diff
@@ -54,49 +55,47 @@ public void keyTyped(KeyEvent e)
 	@Override
 	public void keyPressed(KeyEvent e)
 	{
-		if (!e.isControlDown() || !chatCommandsConfig.clearShortcuts())
+		if (chatCommandsConfig.clearSingleWord() != Keybind.NOT_SET && e.getKeyCode() == chatCommandsConfig.clearSingleWord().getKeyCode())
```

Comment:
I would still keep this as switch() and just compare against config, and then just add early return based on config. will make diff way smaller instead of what you did atm

Reply:
I don't think you can switch key binds can you? Also I am pretty sure for this to work correctly with the modifiers you need to be using `Keybind::matches` and not just comparing the key codes like this.

Reply:
Correct. Using switch didn’t work because the config() stuff isn’t a constant. The way around it would be on every key pressed even would be to assign them to a final or use onChangedConfig. 

Also I tried a few different key binds and they worked but I never tried to purposely do wrong ones to see if they’d match. I’ll see what I can do. Out on vacation currently.

Reply:
You need to use matches to support modifiers correctly. You can never use a switch here because you can't switch on things that are not primitives, enums, or a string. You also don't need to manually check NOT_SET, just do `if (chatCommandsConfig.clearSingleWord().matches(e))`...

Reply:
Yeah my suggestion about switch was mostly because how it was implemented atm, but it needs to use matches instead as adam and abex already said

Reply:
chatCommandsConfig.clearChatBox().matches(e) worked great. Good idea @abextm .